### PR TITLE
handle empty string in _resource_disk_exists

### PR DESCRIPTION
**Pull request by ejarvi**

### CODE REVIEW

File: VMEncryption/main/ResourceDiskUtil.py [hunk 1]
```diff
@@ -86,8 +86,8 @@ def _is_luks_device(self):
         return (int)(self.executor.Execute(cmd, suppress_logging=True)) == CommonVariables.process_success
 
     def _resource_disk_exists(self):
-        """ true if udev name for resource disk exists """
-        cmd = 'test -b ' + self._get_rd_base_dev_path()
+        """ true if udev name for resource disk exists and is mounted """
+        cmd = 'mountpoint -q ' + self._get_rd_base_dev_path()
         if (int)(self.executor.Execute(cmd, suppress_logging=True)) == CommonVariables.process_success:
             self.logger.log("Resource disk exists.")
```
Comment (ShridharParameshwarBhat) [inline VMEncryption/main/ResourceDiskUtil.py:92]: Update in the comment saying "Resource disk exists and is mounted"

File: VMEncryption/main/ResourceDiskUtil.py [hunk 2]
```diff
@@ -86,8 +86,8 @@ def _is_luks_device(self):
         return (int)(self.executor.Execute(cmd, suppress_logging=True)) == CommonVariables.process_success
 
     def _resource_disk_exists(self):
-        """ true if udev name for resource disk exists """
-        cmd = 'test -b ' + self._get_rd_base_dev_path()
+        """ true if udev name for resource disk exists and is mounted """
+        cmd = 'mountpoint -q ' + self._get_rd_base_dev_path()
```
Comment (vimish) [inline VMEncryption/main/ResourceDiskUtil.py:90]: I think self._get_rd_base_dev_path() returns the path to the resource disk itself (like /dev/disk/azure/resource) and not the mountpoint (/mnt/resource).
Will the logic not break in that case?
Correct me if my understanding is wrong here.

Comment (ejarvi) [reply VMEncryption/main/ResourceDiskUtil.py:90]: I think you're right - maybe need to split this into two separate checks:  

1. if base dev path exists
2. if fs is mounted on it

Comment (ejarvi) [reply VMEncryption/main/ResourceDiskUtil.py:90]: Actually the issue here is that self.get_rd_base_dev_path() returns empty string, and test -b on empty string returns 0 

(REPL)
>>> if rdu._get_rd_base_dev_path():
...   print("found")
... else:
...   print("not found")
...
not found

(bash)
[root@ejazwxwmhg02 main]# test -b
[root@ejazwxwmhg02 main]# echo $?
0

Comment (ejarvi) [reply VMEncryption/main/ResourceDiskUtil.py]: added check for empty string

File: VMEncryption/main/ResourceDiskUtil.py (hunk 1 above)
Comment (ejarvi) [reply VMEncryption/main/ResourceDiskUtil.py]: reverted

**Pull request merged**